core/templates/dev/head/directives.js

```diff
@@ -23,9 +23,17 @@ oppia.directive('angularHtmlBind', ['$compile', function($compile) {
   return {
     restrict: 'A',
     link: function(scope, elm, attrs) {
+      // Clean up old scopes if the html changes.
```

Comment:
Interesting! Out of curiosity, why did you need this?

Reply:
For example if the student enters an incorrect answer the interaction "refreshes" which changes the value of angular-html-bind. However if you don't destroy the scope the old callback on 'progress-nav-submit' lingers around, so you now have two callbacks for that event when the student presses the submit button. And the old callback's answer is always processed first, so the student can never correct their answer.

Reply:
You know, that sounds familiar -- at one point we had two calls to submitAnswer() coming in for each submit operation and we couldn't figure out why. I think you might have just solved this long-standing mystery :)